SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestAttachedComponentReplication/SpatialTestAttachedComponentReplication.cpp

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright 1998-2019 Epic Games, Inc. All Rights Reserved.
+
+#include "SpatialTestAttachedComponentReplication.h"
+
+#include "Algo/Copy.h"
+#include "Kismet/GameplayStatics.h"
+#include "Net/UnrealNetwork.h"
+#include "SpatialFunctionalTestStep.h"
+
+ASpatialTestAttachedComponentReplicationActor::ASpatialTestAttachedComponentReplicationActor()
+{
+	bReplicates = true;
+
+	bAlwaysRelevant = true;
+}
+
+USpatialTestAttachedComponentReplicationComponent::USpatialTestAttachedComponentReplicationComponent()
+{
+	SetIsReplicatedByDefault(true);
+}
+
+void USpatialTestAttachedComponentReplicationComponent::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, ReplicatedValue);
+}
+
+ASpatialTestAttachedComponentReplicationActorWithDefaultComponent::ASpatialTestAttachedComponentReplicationActorWithDefaultComponent()
+{
+	CreateDefaultSubobject<USpatialTestAttachedComponentReplicationComponent>(TEXT("DefaultAttachedComponent"));
+}
+
+ASpatialTestAttachedComponentReplication::ASpatialTestAttachedComponentReplication()
+{
+	Author = TEXT("Dmitrii Kozlov <dmitriikozlov@improbable.io>");
```

Comment:
Do you have a testgyms PR for this? Can you link it in the PR description please.

Reply:
Nah, I'll have to add map generation first.

Reply:
Will add once done.

Reply:
Ah I don't need to, do I? I've pushed a34c1c2 which adds map generation setup.